kotlin-inject-compiler/ksp/src/main/kotlin/me/tatarka/inject/compiler/ksp/InjectProcessor.kt

```
@@ -27,63 +28,69 @@ class InjectProcessor(
-            packageName = COMPONENT.packageName,
-            simpleName = COMPONENT.simpleName
-        )
-        deferredClasses = componentSymbols.filterNot { element ->
+        val componentSymbols =
+            resolver.getSymbolsWithAnnotation(COMPONENT.canonicalName).mapNotNull { it as? KSClassDeclaration }
+        val deferredClasses = componentSymbols.filterNot { element ->
             processInject(element, provider, codeGenerator, injectGenerator)
-        }
+        }.toList()
+        deferredClassNames = deferredClasses.mapNotNull { it.qualifiedName }
 
-        val kmpComponentCreateSymbols = previousDeferredFunctions + resolver.getSymbolsWithFunctionAnnotation(
-            packageName = KMP_COMPONENT_CREATE.packageName,
-            simpleName = KMP_COMPONENT_CREATE.simpleName
-        )
-        deferredFunctions = kmpComponentCreateSymbols.filterNot { element ->
+        val kmpComponentCreateSymbols = resolver.getSymbolsWithAnnotation(KMP_COMPONENT_CREATE.canonicalName)
+            .mapNotNull { it as? KSFunctionDeclaration }
+        val deferredFunctions = kmpComponentCreateSymbols.filterNot { element ->
             processKmpComponentCreate(element, provider, kmpComponentCreateFunctionsByComponentType)
-        }
+        }.toList()
+        deferredFunctionNames = deferredFunctions.mapNotNull { it.qualifiedName }
 
         return deferredClasses + deferredFunctions
     }
 
     override fun finish() {
-        // Last round, generate as much as we can, reporting errors for types that still can't be resolved.
-        for (element in deferredClasses) {
-            processInject(
-                element,
-                provider,
+        try {
+            // Last round, generate as much as we can, reporting errors for types that still can't be resolved.
+            val resolver = lastResolver ?: return
+            for (element in deferredClassNames.mapNotNull { resolver.getClassDeclarationByName(it) }) {
```

Same here, I would not silently drop classes.